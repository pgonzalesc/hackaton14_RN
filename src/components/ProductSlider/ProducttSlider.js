import React, {useState, useEffect} from 'react';
import {
  View,
  Text,
  StyleSheet,
  Dimensions,
  FlatList,
  Animated,
} from 'react-native';
import ProductSliderItem from './ProductSliderItem';

const {height, width} = Dimensions.get('window');

const styles = StyleSheet.create({
  dot: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
});

const AnimatedFlatList = Animated.createAnimatedComponent(FlatList);

const ProductSlider = ({navigation, data}) => {
  const scrollX = new Animated.Value(0);

  const detail = (item) => {
    navigation.navigate('Detalle', item);
  };

  if (data && data.length) {
    return (
      <View style={{flex: 1}}>
        <AnimatedFlatList
          data={data}
          keyExtractor={(item, index) => 'key' + index}
          horizontal
          scrollEnabled
          snapToAlignment="center"
          scrollEventThrottle={16}
          decelerationRate="fast"
          showsHorizontalScrollIndicator={false}
          renderItem={(item) => {
            return <ProductSliderItem item={item.item} onPress={()=>detail(item.item)}/>;
          }}
          onScroll={Animated.event(
            [{nativeEvent: {contentOffset: {x: scrollX}}}],
            {
                //listener: (event) => console.log(event),
                useNativeDriver: true,
            },
          )}
        />
      </View>
    );
  }
  return null;
};

export default ProductSlider;
