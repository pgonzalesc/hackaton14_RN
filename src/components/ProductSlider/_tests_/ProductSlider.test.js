import React from 'react';
import { } from 'react-native';
import {shallow} from 'enzyme';
import ProductSlider from '../ProducttSlider';
import {productSliderData} from '../../../data/productSliderData';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('Pruebas en <ProductSlider/>', () => {
    let wrapper = shallow(<ProductSlider data={productSliderData}/>);

    beforeEach(()=>{
        wrapper = shallow(<ProductSlider data={productSliderData}/>);
    });

    test('Debe de mostrar <ProductSlider/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de renderizar', () => {
        expect(wrapper.exists()).toBe(true);
    });

    test('Debe de coincidir los props', () => {
        const data = wrapper.props().children.props.data;
        expect(data).toEqual(productSliderData);
    });
    
});
