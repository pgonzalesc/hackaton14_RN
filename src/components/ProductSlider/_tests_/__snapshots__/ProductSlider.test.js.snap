// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ProductSlider/> Debe de mostrar <ProductSlider/> correctamente 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ForwardRef(AnimatedComponentWrapper)
    data={
      Array [
        Object {
          "id": 1,
          "image": "https://www.starbucks.com.pe/media/cappucino-espresso_tcm92-18666_w1024_n.png",
          "price": "$80",
          "title": "Cappuccino Espresso",
        },
        Object {
          "id": 2,
          "image": "https://www.starbucks.com.pe/media/cafe-mocha-espresso_tcm92-18665_w1024_n.png",
          "price": "$60",
          "title": "Café Mocha",
        },
        Object {
          "id": 3,
          "image": "https://www.starbucks.com.pe/media/mocha-blanco-espresso_tcm92-18664_w1024_n.png",
          "price": "$70",
          "title": "Café Mocha Blanco",
        },
        Object {
          "id": 4,
          "image": "https://www.starbucks.com.pe/media/caramel-macchiato-espresso_tcm92-18662_w1024_n.png",
          "price": "$45",
          "title": "Caramel Macchiato",
        },
      ]
    }
    decelerationRate="fast"
    horizontal={true}
    keyExtractor={[Function]}
    onScroll={[Function]}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={16}
    showsHorizontalScrollIndicator={false}
    snapToAlignment="center"
  />
</View>
`;
